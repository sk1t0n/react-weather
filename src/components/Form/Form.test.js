import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Form } from './index';

it('renders correctly form component', () => {
  const { asFragment } = render(<Form />);
  expect(asFragment()).toMatchSnapshot();
});

describe('Testing DOM', () => {
  global.fetch = jest.fn();

  afterEach(() => {
    fetch.mockClear();
  });

  it('submit valid data', async () => {
    global.fetch = jest.fn(() => Promise.resolve({
      ok: true,
      json: () => Promise.resolve({ temp_c: 0, temp_f: 32 })
    }));

    render(<Form />);
    userEvent.type(
      screen.getByPlaceholderText('Enter city'),
      'Moscow'
    );
    await waitFor(() => {
      userEvent.click(screen.getByText('Submit'));
    });

    expect(screen.getByRole('alert')).toHaveTextContent('Temperature');
  });

  it('submit invalid data', async () => {
    global.fetch = jest.fn(() => Promise.resolve({
      ok: false,
      status: '400'
    }));

    render(<Form />);
    userEvent.type(
      screen.getByPlaceholderText('Enter city'),
      'Invalid city'
    );
    await waitFor(() => {
      userEvent.click(screen.getByText('Submit'));
    });

    expect(screen.getByRole('alert')).toHaveTextContent('City not found');
  });

  it('click the reset button', async () => {
    render(<Form />);
    userEvent.type(
      screen.getByPlaceholderText('Enter city'),
      'Moscow'
    );
    
    expect(screen.getByDisplayValue('Moscow')).toBeInTheDocument();
    userEvent.click(screen.getByText('Reset'));
    expect(screen.queryByDisplayValue('Moscow')).toBeNull();
  });
});
