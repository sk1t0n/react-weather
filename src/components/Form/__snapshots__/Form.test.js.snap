// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly form component 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="city"
      >
        City
      </label>
      <input
        class="form-control"
        id="city"
        placeholder="Enter city"
        type="text"
      />
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Submit
    </button>
    <button
      class="mx-3 btn btn-warning"
      type="reset"
    >
      Reset
    </button>
    <div
      class="mt-5"
    />
  </form>
</DocumentFragment>
`;
